src/app/event-songs/so-event-songs.js

```diff
@@ -20,6 +20,7 @@ function SoEventSongsDirective() {
                 $scope.$apply();
 
                 $rootScope.notifyActivityStream(msg);
+                votingService.setVotesongsOfQueue(msg.nextSongs);
```

Comment:
have you tested it, that the icon gets updated if the music_changed event occurs? maybe the function call needs to be before the $scope.$apply() call, to update the icons correctly.